<a href="https://colab.research.google.com/github/almajed-f/es2245/blob/main/Copy_of_ES2245_lab1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Lab Assignment 1: Introduction to Python for Geoscientists

**Course:** ES2245, The Ohio State University

**Topic:** Variables, Data Structures, and Basic Logic  

**Time needed:** ~60 Minutes  

## Objective
By the end of this lab, you should be able to:
1. Create and manipulate Python variables and lists.
2. Perform basic mathematical calculations on geoscience data.
3. Use `for` loops to iterate through datasets.
4. Use `if/else` statements to classify data.

### Part 1: Variables and Unit Conversions (10 mins)

**Scenario:** You are analyzing borehole temperature data. The drilling team has provided depths in **feet**, but your models require **meters**.

**Task:**
1. Create a variable named `depth_ft` and assign it the value `3500`.
2. Create a variable named `conversion_factor` and assign it the value `0.3048` (1 foot = 0.3048 meters).
3. Calculate the depth in meters and assign it to a new variable `depth_m`.
4. Print a formatted sentence stating the result (e.g., *"The borehole depth is X meters"*).

In [ ]:
depth_ft = 3500
conversion_factor = 0.3048

# Calculate depth_m
depth_m = depth_ft * conversion_factor

# Print result the borehole depth is X meters
print(f"The borehole depth is {depth_m} meters")

The borehole depth is 1066.8 meters


### Part 2: Working with Lists (15 mins)

**Scenario:** You have a list of earthquake magnitudes recorded by a local seismometer over the last week.

**Data:** `[2.4, 3.1, 1.8, 4.5, 2.9, 0.5, 3.3]`

**Task:**
1. Create a list named `magnitudes` containing the data above.
2. Print the **first** magnitude in the list.
3. Print the **last** magnitude in the list (try using negative indexing).
4. Append a new reading of `5.1` to the list.
5. Print the total number of recordings (length of the list).

In [ ]:
magnitudes = [2.4, 3.1, 1.8, 4.5, 2.9, 0.5, 3.3]

# Print first element
print(f"First magnitude: {magnitudes[0]}")

# Print last element
print(f"Last magnitude: {magnitudes[-1]}")

# Append 5.1
magnitudes.append(5.1)
print(f"Magnitudes after appending 5.1: {magnitudes}")

# Print length of list
print(f"Total number of recordings: {len(magnitudes)}")

First magnitude: 2.4
Last magnitude: 3.3
Magnitudes after appending 5.1: [2.4, 3.1, 1.8, 4.5, 2.9, 0.5, 3.3, 5.1]
Total number of recordings: 8


### Part 3: Loops and Math (20 mins)

**Scenario:** You need to calculate the average magnitude of the earthquakes from Part 2. While Python has libraries for this, it is important to know how to do it manually first.

**Task:**
1. Initialize a variable `total_magnitude` to `0`.
2. Write a `for` loop that iterates through your `magnitudes` list.
3. Inside the loop, add each magnitude to `total_magnitude`.
4. After the loop, calculate the mean (average) by dividing the total by the number of earthquakes.
5. Print the average.

In [ ]:
total_magnitude = 0

# Write your for loop here:
for mag in magnitudes:
    total_magnitude += mag

# Calculate mean
average_magnitude = total_magnitude / len(magnitudes)

# Print result
print(f"The average earthquake magnitude is: {average_magnitude:.2f}")

The average earthquake magnitude is: 2.95


### Part 4: Conditional Logic (15 mins)

**Scenario:** You need to classify the earthquakes into categories to alert the public.
* **Magnitude < 3.0:** "Micro" (Not usually felt)
* **Magnitude >= 3.0:** "Minor" (Often felt)

**Task:**
1. Write a loop that goes through the `magnitudes` list.
2. Inside the loop, use an `if/else` statement to check the magnitude.
3. Print the magnitude and its classification (e.g., *"Magnitude 4.5 is Minor"*).

In [ ]:
# Part 4: Your Code Here

for mag in magnitudes:
    if mag < 3.0:
        pass # Replace with print statement
    else:
        pass # Replace with print statement

### Part 5: Putting it all together




**Scenario:** Geothermal gradient calculation.
The temperature at the surface is 15°C. The temperature increases by 25°C for every kilometer of depth (Geothermal Gradient).

**Task:**
1. Create a list of depths in km: `[0, 1, 2, 3, 4, 5]`.
2. Use a loop to calculate the temperature at each depth.
3. Store these temperatures in a new list called `temperatures`.
4. Print the final list.

In [ ]:
# Part 5: Your Code Here

depths = [0, 1, 2, 3, 4, 5]
temperatures = []

surface_temp = 15
gradient = 25

# Loop through depths, calculate temp, and append to temperatures list


# Task
Complete the Python lab assignment by: 1) converting `depth_ft` (3500) to `depth_m` using `conversion_factor` (0.3048) and printing the result; 2) creating a `magnitudes` list `[2.4, 3.1, 1.8, 4.5, 2.9, 0.5, 3.3]`, printing its first and last elements, appending `5.1`, and printing its length; 3) calculating and printing the average of `magnitudes` using a `for` loop; 4) looping through `magnitudes` and classifying each as "Micro" (if < 3.0) or "Minor" (if >= 3.0); and 5) calculating temperatures at depths `[0, 1, 2, 3, 4, 5]` given a surface temperature of 15°C and a gradient of 25°C/km, storing them in a `temperatures` list, and printing the list.

## Complete Part 1: Variables and Unit Conversions

### Subtask:
Create variables for depth in feet and conversion factor, calculate depth in meters, and print the result as a formatted sentence.


**Reasoning**:
I will complete the code in `0tjtA0WSxGGO` by calculating `depth_m` and then printing the formatted result as instructed.



In [ ]:
depth_ft = 3500
conversion_factor = 0.3048

# Calculate depth_m
depth_m = depth_ft * conversion_factor

# Print result the borehole depth is X meters
print(f"The borehole depth is {depth_m} meters")

The borehole depth is 1066.8 meters


## Complete Part 2: Working with Lists

### Subtask:
Create a list of earthquake magnitudes, print the first and last magnitudes, append a new magnitude, and print the total number of recordings.


**Reasoning**:
I will complete the code in `m0LF2inBxGGQ` by performing the requested operations on the `magnitudes` list: printing the first and last elements, appending a new value, and then printing the length of the updated list.



In [ ]:
magnitudes = [2.4, 3.1, 1.8, 4.5, 2.9, 0.5, 3.3]

# Print first element
print(f"First magnitude: {magnitudes[0]}")

# Print last element
print(f"Last magnitude: {magnitudes[-1]}")

# Append 5.1
magnitudes.append(5.1)
print(f"Magnitudes after appending 5.1: {magnitudes}")

# Print length of list
print(f"Total number of recordings: {len(magnitudes)}")

First magnitude: 2.4
Last magnitude: 3.3
Magnitudes after appending 5.1: [2.4, 3.1, 1.8, 4.5, 2.9, 0.5, 3.3, 5.1]
Total number of recordings: 8


## Complete Part 3: Loops and Math

### Subtask:
Initialize a variable for total magnitude, use a `for` loop to sum the magnitudes, calculate the average magnitude, and print the average.


I will complete the code in `63-qTol-xGGR` by implementing the `for` loop to sum the magnitudes, then calculating the average magnitude and printing the result as instructed.

**Reasoning**:
I will complete the code in the provided cell `63-qTol-xGGR` by implementing the `for` loop to sum the magnitudes, calculating the average magnitude using the `len()` function, and then printing the calculated average.



In [ ]:
total_magnitude = 0

# Write your for loop here:
for mag in magnitudes:
    total_magnitude += mag

# Calculate mean
average_magnitude = total_magnitude / len(magnitudes)

# Print result
print(f"The average earthquake magnitude is: {average_magnitude:.2f}")

The average earthquake magnitude is: 2.95


## Complete Part 4: Conditional Logic

### Subtask:
Loop through the magnitudes list and use an `if/else` statement to classify and print each magnitude as 'Micro' or 'Minor'.


**Reasoning**:
I will complete the code in the provided cell `51q8fgMmxGGS` by implementing a `for` loop with an `if/else` statement to classify each earthquake magnitude as 'Micro' or 'Minor' and print the result.



In [ ]:
for mag in magnitudes:
    if mag < 3.0:
        print(f"Magnitude {mag} is Micro")
    else:
        print(f"Magnitude {mag} is Minor")

Magnitude 2.4 is Micro
Magnitude 3.1 is Minor
Magnitude 1.8 is Micro
Magnitude 4.5 is Minor
Magnitude 2.9 is Micro
Magnitude 0.5 is Micro
Magnitude 3.3 is Minor
Magnitude 5.1 is Minor


## Complete Part 5: Geothermal Gradient Calculation

### Subtask:
Create a list of depths, loop through the depths to calculate the temperature at each depth based on the given surface temperature and gradient, and store these temperatures in a new list.


**Reasoning**:
I will complete the code in the provided cell `H-RVr2OjxGGT` by implementing the `for` loop to calculate the temperature at each depth using the given formula, appending the results to the `temperatures` list, and then printing the final `temperatures` list.



In [ ]:
depths = [0, 1, 2, 3, 4, 5]
temperatures = []

surface_temp = 15
gradient = 25

# Loop through depths, calculate temp, and append to temperatures list
for depth in depths:
    current_temp = surface_temp + (depth * gradient)
    temperatures.append(current_temp)

print(f"Temperatures at various depths: {temperatures}")

Temperatures at various depths: [15, 40, 65, 90, 115, 140]


## Final Task

### Subtask:
Review all completed sections and ensure all objectives of the lab assignment have been met.


## Summary:

### Data Analysis Key Findings

*   **Unit Conversion**: A depth of 3500 feet was successfully converted to 1066.8 meters using a conversion factor of 0.3048.
*   **List Manipulation**: The initial `magnitudes` list `[2.4, 3.1, 1.8, 4.5, 2.9, 0.5, 3.3]` was processed; its first element was 2.4 and its last was 3.3. After appending 5.1, the list's length became 8.
*   **Average Calculation**: The average earthquake magnitude from the updated list was calculated to be 2.95.
*   **Magnitude Classification**: Each magnitude was classified: values below 3.0 were categorized as "Micro" (e.g., 2.4, 1.8, 2.9, 0.5), and values 3.0 or greater were classified as "Minor" (e.g., 3.1, 4.5, 3.3, 5.1).
*   **Temperature Gradient Calculation**: For depths `[0, 1, 2, 3, 4, 5]` km, with a surface temperature of 15°C and a gradient of 25°C/km, the corresponding temperatures were calculated as `[15, 40, 65, 90, 115, 140]`°C.

### Insights or Next Steps

*   The successfully completed tasks demonstrate foundational Python programming skills, including variable assignment, list operations, looping, conditional logic, and basic arithmetic, which are essential for geological data analysis.
*   Future analysis could involve visualizing these calculated data points (e.g., magnitude distribution, temperature-depth profile) to gain further insights or integrating these calculations into more complex models, such as seismic hazard assessment or geothermal energy potential.


# Task
All parts of the Python lab assignment, including variables and unit conversions, working with lists, loops and math, conditional logic, and geothermal gradient calculation, have been completed and reviewed.

## Complete Part 1: Variables and Unit Conversions

### Subtask:
Create variables for depth in feet and conversion factor, calculate depth in meters, and print the result as a formatted sentence.


**Reasoning**:
The subtask requires running the provided code to perform unit conversion and print the result. I will execute the code cell that already contains the necessary Python code.



In [ ]:
depth_ft = 3500
conversion_factor = 0.3048

# Calculate depth_m
depth_m = depth_ft * conversion_factor

# Print result the borehole depth is X meters
print(f"The borehole depth is {depth_m} meters")

The borehole depth is 1066.8 meters


## Complete Part 2: Working with Lists

### Subtask:
Create a list of earthquake magnitudes, print the first and last magnitudes, append a new magnitude, and print the total number of recordings.


**Reasoning**:
I will complete the code in the provided cell `m0LF2inBxGGQ` by adding statements to print the first and last elements of the `magnitudes` list, append a new value `5.1` to it, and then print the updated length of the list.



In [ ]:
magnitudes = [2.4, 3.1, 1.8, 4.5, 2.9, 0.5, 3.3]

# Print first element
print(f"First magnitude: {magnitudes[0]}")

# Print last element
print(f"Last magnitude: {magnitudes[-1]}")

# Append 5.1
magnitudes.append(5.1)
print(f"Magnitudes after appending 5.1: {magnitudes}")

# Print length of list
print(f"Total number of recordings: {len(magnitudes)}")

First magnitude: 2.4
Last magnitude: 3.3
Magnitudes after appending 5.1: [2.4, 3.1, 1.8, 4.5, 2.9, 0.5, 3.3, 5.1]
Total number of recordings: 8


## Complete Part 3: Loops and Math

### Subtask:
Initialize a variable for total magnitude, use a `for` loop to sum the magnitudes, calculate the average magnitude, and print the average.


**Reasoning**:
I will complete the code in the provided cell `63-qTol-xGGR` by implementing the `for` loop to sum the magnitudes, calculating the average magnitude using the `len()` function, and then printing the calculated average.



In [ ]:
total_magnitude = 0

# Write your for loop here:
for mag in magnitudes:
    total_magnitude += mag

# Calculate mean
average_magnitude = total_magnitude / len(magnitudes)

# Print result
print(f"The average earthquake magnitude is: {average_magnitude:.2f}")

The average earthquake magnitude is: 2.95


## Complete Part 4: Conditional Logic

### Subtask:
Loop through the magnitudes list and use an `if/else` statement to classify and print each magnitude as 'Micro' or 'Minor'.


**Reasoning**:
I will complete the code in the provided cell `51q8fgMmxGGS` by implementing a `for` loop with an `if/else` statement to classify each earthquake magnitude as 'Micro' or 'Minor' and print the result.



In [ ]:
for mag in magnitudes:
    if mag < 3.0:
        print(f"Magnitude {mag} is Micro")
    else:
        print(f"Magnitude {mag} is Minor")

Magnitude 2.4 is Micro
Magnitude 3.1 is Minor
Magnitude 1.8 is Micro
Magnitude 4.5 is Minor
Magnitude 2.9 is Micro
Magnitude 0.5 is Micro
Magnitude 3.3 is Minor
Magnitude 5.1 is Minor


## Complete Part 5: Geothermal Gradient Calculation

### Subtask:
Create a list of depths, loop through the depths to calculate the temperature at each depth based on the given surface temperature and gradient, and store these temperatures in a new list.


**Reasoning**:
I will complete the code in the provided cell `H-RVr2OjxGGT` by implementing the `for` loop to calculate the temperature at each depth using the given formula, appending the results to the `temperatures` list, and then printing the final `temperatures` list.



In [ ]:
depths = [0, 1, 2, 3, 4, 5]
temperatures = []

surface_temp = 15
gradient = 25

# Loop through depths, calculate temp, and append to temperatures list
for depth in depths:
    current_temp = surface_temp + (depth * gradient)
    temperatures.append(current_temp)

print(f"Temperatures at various depths: {temperatures}")

Temperatures at various depths: [15, 40, 65, 90, 115, 140]


## Final Task

### Subtask:
Review all completed sections and ensure all objectives of the lab assignment have been met.


## Summary:

### Data Analysis Key Findings

*   **Variables and Unit Conversions**: A borehole depth of 3500 feet was successfully converted to 1066.8 meters using a conversion factor of 0.3048.
*   **Working with Lists**: An initial list of earthquake magnitudes `[2.4, 3.1, 1.8, 4.5, 2.9, 0.5, 3.3]` was processed. The first magnitude was 2.4, and the last was 3.3. After appending 5.1, the list contained 8 recordings.
*   **Loops and Math**: The average magnitude from the updated list was calculated to be 2.95 (formatted to two decimal places).
*   **Conditional Logic**: Each magnitude was classified as 'Micro' (less than 3.0) or 'Minor' (3.0 or greater). For example, 2.4 was classified as 'Micro', while 3.1 was classified as 'Minor'.
*   **Geothermal Gradient Calculation**: For a surface temperature of 15°C and a geothermal gradient of 25°C/unit depth, temperatures at depths 0 through 5 were calculated as `[15, 40, 65, 90, 115, 140]` respectively.

### Insights or Next Steps

*   All parts of the Python lab assignment were successfully completed, demonstrating fundamental programming concepts including variable manipulation, list operations, iterative processing, conditional logic, and basic scientific calculations.
*   To further enhance the analysis, consider integrating these separate functionalities into a more comprehensive program or function, for instance, by creating a class to manage earthquake data, including classification and statistical analysis methods.


# Task
The entire task has been completed.


## Complete Part 1: Variables and Unit Conversions

### Subtask:
Create variables for depth in feet and conversion factor, calculate depth in meters, and print the result as a formatted sentence.


**Reasoning**:
The code in cell `0tjtA0WSxGGO` already performs all the steps required for this subtask: creating the variables, calculating `depth_m`, and printing the formatted result. Executing this cell will complete the subtask.



In [ ]:
depth_ft = 3500
conversion_factor = 0.3048

# Calculate depth_m
depth_m = depth_ft * conversion_factor

# Print result the borehole depth is X meters
print(f"The borehole depth is {depth_m} meters")

The borehole depth is 1066.8 meters


## Complete Part 2: Working with Lists

### Subtask:
Create a list of earthquake magnitudes, print the first and last magnitudes, append a new magnitude, and print the total number of recordings.


**Reasoning**:
I will complete the code in the provided cell `m0LF2inBxGGQ` by adding statements to print the first and last elements of the `magnitudes` list, append a new value `5.1` to it, and then print the updated length of the list.

**Reasoning**:
The current subtask requires executing the Python code to perform operations on the `magnitudes` list, which includes printing the first and last elements, appending a new value, and then printing the length of the list. The provided code in cell `m0LF2inBxGGQ` already implements these steps.



In [ ]:
magnitudes = [2.4, 3.1, 1.8, 4.5, 2.9, 0.5, 3.3]

# Print first element
print(f"First magnitude: {magnitudes[0]}")

# Print last element
print(f"Last magnitude: {magnitudes[-1]}")

# Append 5.1
magnitudes.append(5.1)
print(f"Magnitudes after appending 5.1: {magnitudes}")

# Print length of list
print(f"Total number of recordings: {len(magnitudes)}")

First magnitude: 2.4
Last magnitude: 3.3
Magnitudes after appending 5.1: [2.4, 3.1, 1.8, 4.5, 2.9, 0.5, 3.3, 5.1]
Total number of recordings: 8


## Complete Part 3: Loops and Math

### Subtask:
Initialize a variable for total magnitude, use a `for` loop to sum the magnitudes, calculate the average magnitude, and print the average.


**Reasoning**:
I will complete the code in the provided cell by initializing `total_magnitude`, implementing a `for` loop to sum the magnitudes, calculating the average magnitude using `len()`, and then printing the calculated average formatted to two decimal places.



In [ ]:
total_magnitude = 0

# Write your for loop here:
for mag in magnitudes:
    total_magnitude += mag

# Calculate mean
average_magnitude = total_magnitude / len(magnitudes)

# Print result
print(f"The average earthquake magnitude is: {average_magnitude:.2f}")

The average earthquake magnitude is: 2.95


## Complete Part 4: Conditional Logic

### Subtask:
Loop through the magnitudes list and use an an `if/else` statement to classify and print each magnitude as 'Micro' or 'Minor'.


**Reasoning**:
I will complete the code in the provided cell `51q8fgMmxGGS` by replacing the `pass` statements with print statements to classify each earthquake magnitude as 'Micro' or 'Minor' based on the given conditions.



In [ ]:
for mag in magnitudes:
    if mag < 3.0:
        print(f"Magnitude {mag} is Micro") # Replace with print statement
    else:
        print(f"Magnitude {mag} is Minor") # Replace with print statement

Magnitude 2.4 is Micro
Magnitude 3.1 is Minor
Magnitude 1.8 is Micro
Magnitude 4.5 is Minor
Magnitude 2.9 is Micro
Magnitude 0.5 is Micro
Magnitude 3.3 is Minor
Magnitude 5.1 is Minor


## Complete Part 5: Geothermal Gradient Calculation

### Subtask:
Create a list of depths, loop through the depths to calculate the temperature at each depth based on the given surface temperature and gradient, and store these temperatures in a new list.


**Reasoning**:
The subtask requires implementing a `for` loop to calculate temperatures at various depths and then printing the resulting list. The provided code cell `H-RVr2OjxGGT` already contains the necessary initializations, so I will complete the loop and the print statement within that cell.



depths = [0, 1, 2, 3, 4, 5]
temperatures = []

surface_temp = 15
gradient = 25

# Loop through depths, calculate temp, and append to temperatures list
for depth in depths:
    current_temp = surface_temp + (depth * gradient)
    temperatures.append(current_temp)

print(f"Temperatures at various depths: {temperatures}")